### PR TITLE
Added Makefiles in demos/cms,pkcs12, smime directories. Some certificates had expired. Added new ones from test/smime-certs/ directory

### PR DESCRIPTION
**Pull request by meenavyasoracle**
Added new Makefiles in demos/evp, demos/cms, demos/pkcs12 and demos/smime directories.
Some certificates had expired.  Copied certs from test/smime-certs/ directory.

### CODE REVIEW

File: demos/cms/cms_ddec.c [hunk 1]
```diff
@@ -27,7 +27,7 @@ int main(int argc, char **argv)
     ERR_load_crypto_strings();
 
     /* Read in recipient certificate and private key */
-    tbio = BIO_new_file("signer.pem", "r");
+    tbio = BIO_new_file("smrsa1.pem", "r");
```
Comment (richsalz) [inline demos/cms/cms_ddec.c]: remove the blank line.  Don't use ! for pointer, use NULL check

File: demos/cms/cms_ddec.c [hunk 2]
```diff
@@ -68,6 +68,9 @@ int main(int argc, char **argv)
     if (!CMS_decrypt(cms, rkey, rcert, dcont, out, 0))
         goto err;
 
+    printf("Successfully dencrypted contents of file smencr.pem"
```
Comment (richsalz) [inline demos/cms/cms_ddec.c]: there probably need to be some newline characters here.

File: demos/cms/cms_dec.c [hunk 3]
```diff
@@ -24,7 +24,7 @@ int main(int argc, char **argv)
     ERR_load_crypto_strings();
 
     /* Read in recipient certificate and private key */
-    tbio = BIO_new_file("signer.pem", "r");
+    tbio = BIO_new_file("smrsa1.pem", "r");
```
Comment (richsalz) [inline demos/cms/cms_dec.c]: remove the newline, check against NULL

File: demos/cms/cms_dec.c [hunk 4]
```diff
@@ -59,6 +59,9 @@ int main(int argc, char **argv)
     if (!CMS_decrypt(cms, rkey, rcert, NULL, out, 0))
         goto err;
 
+    printf("Successfully dencrypted contents of file smencr.txt into file "
```
Comment (richsalz) [inline demos/cms/cms_dec.c]: newlines.

File: demos/cms/cms_denc.c [hunk 5]
```diff
@@ -29,7 +29,7 @@ int main(int argc, char **argv)
     ERR_load_crypto_strings();
 
     /* Read in recipient certificate */
-    tbio = BIO_new_file("signer.pem", "r");
+    tbio = BIO_new_file("smrsa1.pem", "r");
```
Comment (richsalz) [inline demos/cms/cms_denc.c]: ditto.

File: demos/cms/cms_denc.c [hunk 6]
```diff
@@ -77,6 +77,9 @@ int main(int argc, char **argv)
     if (!PEM_write_bio_CMS(out, cms))
         goto err;
 
+    printf("Successfully encrypted contents of file encr.txt into files"
```
Comment (richsalz) [inline demos/cms/cms_denc.c]: ditto

File: demos/evp/Makefile [hunk 7]
```diff
@@ -18,3 +18,6 @@ aesgcm: aesgcm.o
 
 aesccm aesgcm:
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $<
+
+clean:
+	$(RM) aesccm.o aesgcm.o aesccm aesgcm
```
Comment (richsalz) [inline demos/evp/Makefile]: missing newline at end of line.

Comment (richsalz) [inline demos/evp/Makefile]: you must be an emacs user :)

File: demos/cms/cms_ddec.c (hunk 2 above)
Comment (FdaSilvaYY) [inline demos/cms/cms_ddec.c]: `dencrypted `?  `decrypted` or `unencrypted` may be ...

Comment (meenavyasoracle) [inline demos/cms/cms_ddec.c]: Please tell me what do you mean by "there probably need to be some newline characters here."? I am fixing it right now.

File: demos/evp/Makefile (hunk 7 above)
Comment (meenavyasoracle) [inline demos/evp/Makefile]: No I use vim and cscope on Linux and/or Solaris. 
You mean add a newline like this ?

$git diff Makefile
diff --git a/demos/evp/Makefile b/demos/evp/Makefile
index ebf5cf1..ffa8acc 100644
--- a/demos/evp/Makefile
+++ b/demos/evp/Makefile
@@ -21,3 +21,4 @@ aesccm aesgcm:
 
 clean:
        $(RM) aesccm.o aesgcm.o aesccm aesgcm
+

File: demos/cms/cms_ddec.c (hunk 2 above)
Comment (richsalz) [inline demos/cms/cms_ddec.c]: The output should be <= 78 characters.

**Comment by richsalz** [comment]
Git complained that the file didn’t end with a newline.  I don't know how it happened.

**Comment by richsalz** [comment]
Thanks!

**Pull request closed without merging**